lambdas/access_counts/index.py

```diff
@@ -0,0 +1,200 @@
+"""
+Lambda function that runs Athena queries over CloudTrail logs and .quilt/named_packages/
+and creates summaries of object and package access events.
+"""
+
+import os
+import textwrap
+import time
+
+import boto3
+
+ATHENA_DATABASE = os.environ['ATHENA_DATABASE']
+# Bucket where query results will be stored.
+QUERY_RESULT_BUCKET = os.environ['QUERY_RESULT_BUCKET']
+# A temporary directory where Athena query results will be written.
+QUERY_TEMP_DIR = os.environ['QUERY_TEMP_DIR']
+# Directory where the summary files will be stored.
+ACCESS_COUNTS_OUTPUT_DIR = os.environ['ACCESS_COUNTS_OUTPUT_DIR']
+
+
+def sql_escape(s):
+    return s.replace("'", "''")
+
+
+DROP_OBJECT_ACCESS_LOG = """DROP TABLE IF EXISTS object_access_log"""
+DROP_PACKAGE_HASHES = """DROP TABLE IF EXISTS package_hashes"""
+
+CREATE_OBJECT_ACCESS_LOG = textwrap.dedent(f"""\
+    CREATE TABLE object_access_log
+    WITH (
+        format = 'Parquet',
+        parquet_compression = 'SNAPPY',
+        external_location = 's3://{sql_escape(QUERY_RESULT_BUCKET)}/{sql_escape(QUERY_TEMP_DIR)}/object_access_log/'
+    )
+    AS
+    SELECT eventname, date_format(from_iso8601_timestamp(eventtime), '%Y-%m-%d') AS date, bucket, key
+    FROM (
+        SELECT
+            eventname,
+            eventtime,
+            json_extract_scalar(requestparameters, '$.bucketName') AS bucket,
+            json_extract_scalar(requestparameters, '$.key') AS key
+        FROM cloudtrail
+    )
+    WHERE bucket IS NOT NULL AND key IS NOT NULL
+""")
+
+CREATE_PACKAGE_HASHES = textwrap.dedent(f"""\
+    CREATE TABLE package_hashes
+    WITH (
+        format = 'Parquet',
+        parquet_compression = 'SNAPPY',
+        external_location = 's3://{sql_escape(QUERY_RESULT_BUCKET)}/{sql_escape(QUERY_TEMP_DIR)}/package_hashes/'
+    )
+    AS
+    SELECT
+        split_part("$path", '/', 3) AS bucket,
```

Comment:
comment with expected path, e.g. `s3://BUCKET/AWSLogs/730278974607/CloudTrail/us-east-1/2019/04/01/730278974607_CloudTrail_us-east-1_20190401T0220Z_ZLJUquHVvB845TeS.json.gz` will make these splits easier to read and maintain.

Reply:
Ok